src/System.Runtime.Extensions/src/System/Environment.cs

```diff
@@ -0,0 +1,217 @@
+
+        public static void Exit(int exitCode) => EnvironmentAugments.Exit(exitCode);
+
+        public static int ExitCode { get { return EnvironmentAugments.ExitCode; } set { EnvironmentAugments.ExitCode = value; } }
+
+        public static void FailFast(string message) => FailFast(message, exception: null);
+
+        public static void FailFast(string message, Exception exception) => EnvironmentAugments.FailFast(message, exception);
+
+        public static string ExpandEnvironmentVariables(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (name.Length == 0)
+            {
+                return name;
+            }
+
+            return ExpandEnvironmentVariablesCore(name);
+        }
+
+        public static string[] GetCommandLineArgs() => EnvironmentAugments.GetCommandLineArgs();
+
+        public static string GetEnvironmentVariable(string variable) => GetEnvironmentVariable(variable, EnvironmentVariableTarget.Process);
+
+        public static string GetEnvironmentVariable(string variable, EnvironmentVariableTarget target) => EnvironmentAugments.GetEnvironmentVariable(variable, target);
+
+        public static IDictionary GetEnvironmentVariables() => GetEnvironmentVariables(EnvironmentVariableTarget.Process);
+
+        public static IDictionary GetEnvironmentVariables(EnvironmentVariableTarget target) => EnvironmentAugments.GetEnvironmentVariables(target);
+
+        public static string GetFolderPath(SpecialFolder folder) => GetFolderPath(folder, SpecialFolderOption.None);
+
+        public static string GetFolderPath(SpecialFolder folder, SpecialFolderOption option)
+        {
+            if (!Enum.IsDefined(typeof(SpecialFolder), folder))
+            {
+                throw new ArgumentOutOfRangeException(nameof(folder), folder, SR.Format(SR.Arg_EnumIllegalVal, folder));
+            }
+
+            if (option != SpecialFolderOption.None && !Enum.IsDefined(typeof(SpecialFolderOption), option))
+            {
+                throw new ArgumentOutOfRangeException(nameof(option), option, SR.Format(SR.Arg_EnumIllegalVal, option));
+            }
+
+            return GetFolderPathCore(folder, option);
+        }
+
+        public static bool HasShutdownStarted => EnvironmentAugments.HasShutdownStarted;
+
+        public static bool Is64BitProcess => IntPtr.Size == 8;
+
+        public static bool Is64BitOperatingSystem => Is64BitProcess || Is64BitOperatingSystemWhen32BitProcess;
+
+        public static void SetEnvironmentVariable(string variable, string value) => SetEnvironmentVariable(variable, value, EnvironmentVariableTarget.Process);
+
+        public static void SetEnvironmentVariable(string variable, string value, EnvironmentVariableTarget target) => EnvironmentAugments.SetEnvironmentVariable(variable, value, target);
```

Comment:
You mean environment variables?  I did this because of the coreclr's cache on Unix for reading all of the environment variables and then ensuring synchronized access to that cache.  We can do similar caching in corefx, but then how would we synchronize access between coreclr's use of environment variables and corefx's? And how would we ensure that any environment variables set by coreclr are then visible to corefx? Or maybe coreclr doesn't do any such setting, and it doesn't need to observe any changes made by corefx, in which case none of this would matter and would could move it without problems?